packages/@ourworldindata/grapher/src/entitySelector/EntitySelector.tsx

```diff
@@ -475,12 +509,19 @@ export class EntitySelector extends React.Component<{
 
     @computed private get title(): string {
         return this.manager.isOnMapTab
-            ? `Select ${this.entityTypePlural}`
+            ? `Select ${this.entityType.plural}`
             : this.manager.canHighlightEntities
-              ? `Select ${this.entityTypePlural}`
+              ? `Select ${this.entityType.plural}`
               : this.manager.canChangeEntity
-                ? `Choose ${a(this.entityType)}`
-                : `Add/remove ${this.entityTypePlural}`
+                ? `Choose ${a(this.entityType.singular)}`
+                : `Add/remove ${this.entityType.plural}`
+    }
+
+    @computed private get searchPlaceholderEntityType(): string {
+        if (this.entityFilter === "all") return this.entityType.singular
+        if (this.entityFilter === "countries") return "country"
+        if (this.entityFilter === "incomeGroups") return "income group"
+        return "continent"
```

Comment:
We hit this case for `historicalCountries` also, displaying "Search for a continent".

This would be a good case to use `match` from `ts-pattern`, to ensure a type-exhaustive mapping.